service/controller/v22/resource/s3bucket/metrics.go

```diff
@@ -0,0 +1,26 @@
+package s3bucket
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	prometheusNamespace = "aws_operator"
+	prometehusSubsystem = "resource_s3bucketv22"
+)
+
+var (
+	s3ObjectsTotal = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: prometheusNamespace,
+			Subsystem: prometehusSubsystem,
+			Name:      "s3_objects_total",
+			Help:      "Total number of objects within a S3 bucket.",
```

Comment:
I am wondering how useful this metric is in general as we only count them when the cluster is being deleted. Further, when the deletion breaks in the middle and we retry, the counter should be reset. Also not this here is a gauge. How should we do this?

Reply:
Actually, was the idea to add to the collector, instead of inline with the resource? That would probably alleviate your concerns.

Reply:
The idea was to extend the collector and then I figured that there is no way to just get the stupid number of objects in a bucket. So we have to list and I thought instead of doing a lot of weird requests only to get a stupid number, we could add the metrics where the code is executed already. But then I figured that this is only executed on deletion and might not be useful at all. Do you think extending the exporter is a good idea still? I am rather 50 50.

Reply:
I think doing it in the collector is better, as we'll then get these metrics over time, not just during deletion.

Reply:
Ok, will rework this next week. Will not make it today.